rotkehlchen/interfaces.py

```diff
@@ -70,20 +72,73 @@ def can_query_history(
         """Checks if it's okay to query historical price"""
 
     @abc.abstractmethod
-    def query_historical_price(
+    def _query_historical_price(
             self,
             from_asset: Asset,
             to_asset: Asset,
             timestamp: Timestamp,
     ) -> Price:
-        """An oracle implements this to return a historical price from_asset to to_asset at time.
+        """Implementation specific historical price query logic.
 
         If no price can be found may raise:
         - PriceQueryUnsupportedAsset
         - NoPriceForGivenTimestamp
         - RemoteError
         """
 
+    @abc.abstractmethod
+    def oracle_source(self) -> HistoricalPriceOracle:
+        """Source enum used as key for storing and retrieving cached prices"""
+
+    @abc.abstractmethod
+    def max_price_age_in_seconds(self) -> int:
+        """Maximum time difference in seconds for which a historical price is considered valid"""
```

Comment:
This is wrong. The name suggests something that it's not and it's misleading. This is not the price age. These are historical prices. It's how far from the requested timestamp in either direction a price is found.

```suggestion
    def max_seconds_distance(self) -> int:
        """Maximum distance in seconds around a given timestamp for which if a price is found it will be considered valid and returned"""
```